### ISSUE TITLE
Language Generation Input

### ISSUE THREAD

**Issue by phSyz** [oxford-cs-deepnlp-2017/practical-3#1]
Hi, I'd like to better understand the input/label format for the LTSM RNN model for language generation. As far as I understand (and correct me if I'm wrong), we have a one-hot-encoding of a word which we feed into the network at each training iteration. My question concerns the training label - do we want a one hot encoding of the next word, or do we want some sort of output word distribution? If so, how would we format these distribution labels that we want to feed into the network? Thanks!

**Comment by bshillingford** [comment]
Are you an Oxford student enrolled in the course? If so, please begin to use the Piazza.

To briefly answer your question, one-hot encodings are inefficient for input to an RNN, as you've seen in practical2. E.g. `nn.Embedding` is more efficient in PyTorch (similar things exist in TF and other frameworks), as the inputs are indices rather than explicitly creating one-hot vectors and multiplying them by a matrix.

In Task 1, the loss function is identical to practical 2. In Task 2, it is an RNN LM, see the lectures for details where this is covered in great detail.
